tiledb/sm/query/readers/aggregators/aggregate_with_count.h

```diff
@@ -81,62 +95,56 @@ class AggregateWithCount {
       auto bitmap_values = input_data.bitmap_data_as<BITMAP_T>();
 
       if (field_info_.is_nullable_) {
-        validity = 0;
         auto validity_values = input_data.validity_data();
 
-        // Process for nullable sums with bitmap.
+        // Process for nullable values with bitmap.
         for (uint64_t c = input_data.min_cell(); c < input_data.max_cell();
              c++) {
           if (validity_values[c] != 0 && bitmap_values[c] != 0) {
-            validity = 1;
-
-            auto value = static_cast<SUM_T>(values[c]);
+            AGG_T value = input_data.value_at<VALUE_T>(c);
             for (BITMAP_T i = 0; i < bitmap_values[c]; i++) {
+              agg_policy.op(value, res, count);
               count++;
-              AggPolicy::op(value, sum);
             }
           }
         }
       } else {
-        // Process for non nullable sums with bitmap.
+        // Process for non nullable values with bitmap.
         for (uint64_t c = input_data.min_cell(); c < input_data.max_cell();
              c++) {
-          auto value = static_cast<SUM_T>(values[c]);
+          AGG_T value = input_data.value_at<VALUE_T>(c);
 
           for (BITMAP_T i = 0; i < bitmap_values[c]; i++) {
+            agg_policy.op(value, res, count);
             count++;
-            AggPolicy::op(value, sum);
           }
         }
       }
     } else {
       if (field_info_.is_nullable_) {
-        validity = 0;
         auto validity_values = input_data.validity_data();
 
-        // Process for nullable sums with no bitmap.
+        // Process for nullable values with no bitmap.
         for (uint64_t c = input_data.min_cell(); c < input_data.max_cell();
              c++) {
           if (validity_values[c] != 0) {
-            validity = 1;
```

Comment:
Yes, the function returns the count (which in the case of nullable attributes means non-null count). It can be used to infer validity from the caller.